24.cpp

```diff
@@ -0,0 +1,35 @@
+#include <iostream>
+struct ListNode {
+  int val;
+  ListNode* next;
+  explicit ListNode(int x) : val(x), next(nullptr) {}
+};
+class Solution {
+ public:
+  ListNode* swapPairs(ListNode* head) {
+    ListNode* work = new ListNode(0);
+    work->next = head;
+    head = work;

```

Comment:
`new ListNode` is unnecessary. Seems `work` is the tail of your result list, if so, rename it to `tail`.